forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js

```diff
@@ -0,0 +1,452 @@
+import React, { useState } from "react";
+import { useSelector } from "react-redux";
+import Drawer from "@material-ui/core/Drawer";
+import Button from "@material-ui/core/Button";
+import List from "@material-ui/core/List";
+import Divider from "@material-ui/core/Divider";
+import { saveFilters } from "../../../../apiManager/services/bpmTaskServices";
+
+export default function CreateNewFilterDrawer() {
+  const [openDrawer, setOpenDrawer] = useState(false);
+  const [filterName, setFilterName] = useState("");
+  const [showUndefinedVariable, setShowUndefinedVariable] = useState(false);
+  const [inputVisibility, setInputVisibility] = useState({});
+  const [definitionKeyId, setDefinitionKeyId] = useState("");
+  const [candidateGroup, setCandidateGroup] = useState([]);
+  const [taskName, setTaskName] = useState("");
+  const [includeAssignedTasks, setIncludeAssignedTasks] = useState(false);
+  const [rows, setRows] = useState([
+    { id: 1000, name: "", label: "", add: true, delete: false },
+  ]);
+  const [newRow, setNewRow] = useState([]);
+  const [permissions, setPermissions] = useState("");
+  const [identifierId, setIdentifierId] = useState("");
+  const [selectUserGroupIcon, setSelectUserGroupIcon] = useState("");
+  const [specificUserGroup, setSpecificUserGroup] = useState("");
+  const userName = useSelector((state) => state.user?.userDetail?.name);
+
+  //Function for collecting the input data
+  const handleSubmit = () => {
+    const data = {
+      name: filterName,
+      criteria: {
+        processDefinitionKey: definitionKeyId,
+        candidateGroup: candidateGroup,
+        assignee: taskName,
+        includeAssignedTasks: false,
+      },
+      variables: [
+        ...rows.map((row) => ({
+          name: row.name,
+          label: row.label,
+        })),
+      ],
+      users: [userName],
+      roles: [selectUserGroupIcon],
+      identifierId: identifierId,
+    };
+    saveFilters(data)
+      .then(() => {
+        toggleDrawer(false);
+        // Clearing the input fields after submission
+        setFilterName("");
+        setShowUndefinedVariable("");
+        setInputVisibility("");
+        setDefinitionKeyId("");
+        setCandidateGroup("");
+        setTaskName("");
+        setIncludeAssignedTasks("");
+        setPermissions("");
+        setIdentifierId("");
+        setSelectUserGroupIcon("");
+        setSpecificUserGroup("");
+        setRows([{ id: 1000, name: "", label: "", add: true, delete: false }]);
```

Comment:
can add a constant variableto store the json and use it rather than defining full json  here, and seems its used as default value multiple places